### PR TITLE
fix(pty): respond to DSR cursor position queries for inline TUI compatibility

### PR DESCRIPTION
**Pull request by cavanaug**
## Summary
- Add PTY-side handling for Device Status Report cursor-position requests (`ESC[6n`) so apps running in Pilotty can receive a valid cursor-position response (`ESC[{row};{col}R`).
- Track cursor position from PTY output stream and emit synthetic DSR replies through the existing PTY input writer path.
- Add focused tests for DSR detection (including chunk boundaries), response coordinate correctness (1-indexed), and end-to-end async PTY response behavior.

## Problem
Apps using inline terminal viewports (e.g. `ratatui` `Viewport::Inline`, similar to Atuin-style overlays) depend on cursor-position queries succeeding during startup/rendering.

In Pilotty sessions, these apps could fail with errors equivalent to:
- `The cursor position could not be read within a normal duration`

Root cause: PTY output was parsed for display/snapshot state but DSR cursor queries (`ESC[6n`) were not answered, so querying apps timed out waiting for terminal response.

## Solution Approach
- Introduce a small DSR parser in PTY read path to detect `ESC[6n` reliably, including split/chunked sequences.
- Keep a cursor tracker (vt100-backed) updated from PTY output.
- On DSR query, enqueue a synthetic response `ESC[{row};{col}R` (1-indexed terminal coordinates) back to child stdin via the existing write channel.
- Keep tracker dimensions synchronized on resize.

## Validation
- `cargo test -p pilotty-cli daemon::pty::tests`
- `cargo test -p pilotty-cli`

Both pass locally.

### CODE REVIEW

File: crates/pilotty-cli/src/daemon/pty.rs [hunk 1]
```diff
@@ -261,6 +340,32 @@ impl AsyncPtyHandle {
...
+                                warn!("Cursor tracker mutex poisoned; using fallback cursor response");
+                                (1, 1)
+                            }
+                        };
+                        let response = format!("\x1b[{};{}R", row, col).into_bytes();
+
+                        for _ in 0..dsr_queries {
+                            if write_tx.blocking_send(response.clone()).is_err() {
+                                debug!("PTY write channel closed while sending DSR response");
+                                break;
```
[the remark this answers is not in the document]
Comment (cavanaug) [reply crates/pilotty-cli/src/daemon/pty.rs]: Changed synthetic DSR enqueue from `blocking_send` to `try_send`. On full channel, we log and drop the synthetic response rather than blocking the reader thread; on closed channel, we stop sending. This keeps PTY output draining even under backpressure.

File: crates/pilotty-cli/src/daemon/pty.rs [hunk 2]
```diff
@@ -8,6 +8,71 @@ use portable_pty::{native_pty_system, Child, CommandBuilder, MasterPty, PtySize}
...
+    }
+
+    fn resize(&mut self, size: TermSize) {
+        self.parser.screen_mut().set_size(size.rows, size.cols);
+    }
+
+    fn process_output(&mut self, bytes: &[u8]) -> usize {
+        self.parser.process(bytes);
+        self.count_dsr_queries(bytes)
+    }
```
[the remark this answers is not in the document]
Comment (cavanaug) [reply crates/pilotty-cli/src/daemon/pty.rs]: Updated `CursorTracker::process_output` to process output incrementally and capture `(row, col)` at each detected `ESC[6n` boundary. It now returns `Vec<(u16, u16)>`, so DSR replies use the cursor position at query time even when trailing bytes exist in the same chunk.

File: crates/pilotty-cli/src/daemon/pty.rs [hunk 3]
```diff
@@ -261,6 +340,32 @@ impl AsyncPtyHandle {
...
+                    };
+
+                    if dsr_queries > 0 {
+                        let (row, col) = match cursor_tracker.lock() {
+                            Ok(tracker) => tracker.cursor_position_one_indexed(),
+                            Err(_) => {
+                                warn!("Cursor tracker mutex poisoned; using fallback cursor response");
+                                (1, 1)
+                            }
+                        };
```
[the remark this answers is not in the document]
Comment (cavanaug) [reply crates/pilotty-cli/src/daemon/pty.rs]: Refactored to a single lock per read chunk. `reader_loop` now calls `process_output` once, receives all DSR boundary positions, and emits responses from that vector. This removes the previous double-lock and keeps parse/position capture consistent.

File: crates/pilotty-cli/src/daemon/pty.rs [hunk 4]
```diff
@@ -509,6 +641,45 @@ mod tests {
...
+        .await;
+
+        let bytes = result.expect("Timed out waiting for DSR response");
+        let output = String::from_utf8_lossy(&bytes);
+        assert!(
+            output.contains("ok:"),
+            "Expected shell to receive DSR response, got: {:?}",
+            output
+        );
+
```
[the remark this answers is not in the document]
Comment (cavanaug) [reply crates/pilotty-cli/src/daemon/pty.rs:718]: Strengthened the async E2E test to validate the actual cursor report. The test now writes a known sequence (`abc` + `ESC[6n` + `xyz`), parses `ok:\x1b[{row};{col}`, and asserts `(row, col) == (1, 4)`.
